pkg/ring/lifecycler.go

```diff
@@ -246,18 +248,25 @@ func (i *Lifecycler) ChangeState(ctx context.Context, state IngesterState) error
 	return <-err
 }
 
-func (i *Lifecycler) getTokens() []uint32 {
+func (i *Lifecycler) getTokens() Tokens {
 	i.stateMtx.Lock()
 	defer i.stateMtx.Unlock()
+	if i.tokens == nil {
```

Comment:
I'm missing the need of this `if`. Isn't `return i.tokens` enough?

Reply:
Lots of missed cleanups with changes in designs :)